src/main/java/com/icthh/xm/uaa/service/UserService.java

```diff
@@ -217,11 +220,17 @@ public User createUser(UserDTO user) {
      */
     @LogicExtensionPoint("ChangeUserRole")
     public Optional<UserDTO> changeUserRole(final UserDTO updatedUser) {
-        Preconditions.checkArgument(StringUtils.isNotEmpty(updatedUser.getRoleKey()), "No roleKey provided");
+        Preconditions.checkArgument(isNotEmpty(updatedUser.getAuthorities()), "No roleKey provided");
+        Preconditions.checkArgument(
+            isNotEmpty(updatedUser.getAuthorities()
+                .stream()
+                .filter(Objects::nonNull)
```

Comment:
please check if you really need to do not filter by nonNull because isNoneBlank should check for null as well.

Reply:
even simpler: do you really need to do all this stream --> collector to list with a filter? is there is a chance to get empty roles in authorities?

Reply:
i have removed redundant filter but kept stream just because userDto object comes from UI and there was appropriate check on `role` object earlier.
If u think stream is still redundant i'll remove it

Reply:
ok, let id be here